src/Components/Products/updatedProducts.js

```diff
@@ -0,0 +1,64 @@
+import { useFilter } from "../../context/context";
+
+
+export const FilterFunction = (products) => {
+	const { filterState, filterDispatch } = useFilter();
+	let updatedProductList = products;
+
+	if (filterState.excludeNotAvailable) {
+		updatedProductList = updatedProductList.filter((item) => item.availableOrNot)
+		console.log(updatedProductList)
+	}
+
+	if (filterState.offerItems) {
+		updatedProductList = updatedProductList.filter((item) => item.offer)
+		console.log(updatedProductList)
+	}
+
+	if (filterState.byCategory) {
+		if (filterState.byCategory === 'All') {
+			updatedProductList = products
+		} else {
+			updatedProductList = updatedProductList.filter((item) => item.categoryName === filterState.byCategory)
+			console.log(updatedProductList)
+		}
+	}
+
+	if (filterState.byPrice) {
```

Comment:
Start using prettier for code inclination as it is hard to read. 
UI is good, functionality is working properly. Good to merge